components/renku_data_services/project/db.py

```diff
@@ -285,9 +263,27 @@ async def update_project(
             has_permission = await self.authz.has_permission(user, resource_type, resource_id, Scope.WRITE)
             if not has_permission:
                 raise errors.ForbiddenError(
-                    message=f"The project cannot be moved because you do not have sufficient permissions with the namespace {ns_slug}"  # noqa: E501
+                    message=f"The project cannot be moved because you do not have sufficient permissions with the namespace {patch.namespace}"  # noqa: E501
                 )
             project.slug.namespace_id = ns.id
+        if patch.visibility is not None:
+            visibility_orm = (
+                project_apispec.Visibility(patch.visibility)
+                if isinstance(patch.visibility, str)
+                else project_apispec.Visibility(patch.visibility.value)
+            )
+            project.visibility = visibility_orm
+        if patch.repositories is not None:
+            project.repositories = [
+                schemas.ProjectRepositoryORM(url=r, project_id=project_id_str, project=project)
+                for r in patch.repositories
+            ]
+            # Trigger update for ``updated_at`` column
+            await session.execute(update(schemas.ProjectORM).where(schemas.ProjectORM.id == project_id).values())
+        if patch.description is not None:
+            project.description = patch.description if patch.description else None
+        if patch.keywords is not None:
```

Comment:
question: I think this would currently work but curious if you tested it. Can keywords be unset by sending an empty list `[]` ?

Reply:
same for description with `""`. other fields can't be empty I think

Reply:
Yes, it works. There is a corresponding test for project keywords. Manual testing also shows it works for the description (sending "" resets the description).